### PR TITLE
Identity initializer with zero padding

### PR DESCRIPTION
**Pull request by RaphaelMeudec**
### Summary

This PR changes the behaviour of `initializers.Identity` to match Tensorflow's behaviour (pad with 0s if matrix is not square)

### Related Issues

#11774 

### PR Overview

- [n] This PR requires new unit tests [y/n] (make sure tests are included) -> tests have been updated but no new tests required
- [y] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse I started tackling [#11774](https://github.com/keras-team/keras/issues/11774) in this PR, but I have some questions on the RNN part. Could you precise what is the "correct behavior" expected for the fused kernels? cc @fchollet

**Comment by gabrieldemarmiesse** [comment]
I don't know really. I'm not very familiar with the matter. I'll let @fchollet come back to you about it since he wrote the Request for contribution.

File: keras/initializers.py [hunk 1]
```diff
@@ -283,17 +282,24 @@ def __call__(self, shape, dtype=None):
...
-        if max(shape) % min(shape) != 0:
-            raise ValueError('Long side should be multiple of short side.')
-
         if shape[0] == shape[1]:
             return self.gain * np.identity(shape[0])
         elif shape[0] > shape[1]:
             return self.gain * np.concatenate(
-                [np.identity(shape[1])] * (shape[0] // shape[1]), axis=0)
+                [
+                    np.identity(shape[1]),
```
Comment (fchollet) [inline keras/initializers.py]: Use the formatting

```
[np.identity(shape[1]),
 np.zeros((shape[0] - shape[1], shape[1]))]
```

Comment (fchollet) [reply keras/initializers.py]: (same below)

File: keras/initializers.py [hunk 2]
```diff
@@ -268,8 +268,7 @@ class Identity(Initializer):
     """Initializer that generates the identity matrix.
 
     Only use for 2D matrices.
-    If the long side of the matrix is a multiple of the short side,
-    multiple identity matrices are concatenated along the long side.
+    If the desired matrix is not square, it pads 0s on the additional rows/columns
```
Comment (fchollet) [inline keras/initializers.py]: 'pads with zeros'

**Comment by RaphaelMeudec** [comment]
@fchollet I took your feedback into consideration. I created a `RecurrentIdentity` initializer which is used instead of the `Identity` for GRU and LSTM. Though, is it necessary to have this behaviour? Can't we keep both initializers distinct and let the user choose? (`identity` is not the default value for kernel initializer)

**Comment by fchollet** [comment]
Sorry if I wasn't clear. I didn't mean you should add a new initializer to the API. Rather, I meant to say that, in `LSTMCell` and `GRUCell`, if the user passes the `identity` initializer AND we are in fused kernel mode (`implementation==2`), you should create on the fly an initializer that concatenates multiple calls to the identity initializer. Somewhat similar to what we do for the bias initializer here: https://github.com/keras-team/keras/blob/master/keras/layers/recurrent.py#L1878-L1886

File: keras/layers/recurrent.py [hunk 3]
```diff
@@ -1267,6 +1267,14 @@ def __init__(self, units,
 
     def build(self, input_shape):
         input_dim = input_shape[-1]
+
+        if type(self.recurrent_initializer).__name__ == 'Identity':
```
Comment (fchollet) [inline keras/layers/recurrent.py]: Use `isinstance` for robustness (since the initializer module is already imported)

Comment (fchollet) [reply keras/layers/recurrent.py]: You should also restrict this behavior to the fused kernel case, for simplicity

Comment (RaphaelMeudec) [reply keras/layers/recurrent.py]: By checking on the input shape ratio given to the build method?

**Comment by RaphaelMeudec** [comment]
@fchollet Not sure what you mean by adding an LSTM test as the `recurrent.LSTM` is already in the `rnn_test` parametrization and therefore `test_rnn_cell_identity_initializer` already tests the LSTM case
